packages/aws-cdk-lib/aws-eks/lib/kubectl-provider.ts

```diff
@@ -135,7 +135,7 @@ export class KubectlProvider extends NestedStack implements IKubectlProvider {
 
     const handler = new lambda.Function(this, 'Handler', {
       code: lambda.Code.fromAsset(path.join(__dirname, 'kubectl-handler')),
-      runtime: lambda.Runtime.PYTHON_3_7,
+      runtime: lambda.Runtime.PYTHON_3_11,
```

Comment:
@pahud re https://github.com/aws/aws-cdk/issues/26994#issuecomment-1706777375, I couldn't think of a reason not to pick the latest runtime.

I have checked that all the existing code is syntactically valid in Python 3.11.

There might be reasons, like regional availability, to pick a less recent version, but I'm unsure how to determine that.